### PR TITLE
helm: Fix large sessionTTL values

### LINKED ISSUE #7325: large `sessionTTL` values do not work in helm chart
## Describe the bug
[A clear and concise description of what the bug is. It should also be clear what the expected outcome is.]

using a large `sessionTTL` value (like the max of `3153600` for example) with the helm chart results in an error like the following when starting the pod:

```
2026-08-18T19:07:28.146244855-06:00 invalid value "3.153600e+06" for flag -session-ttl: parse error
```

...and looking at the args on the container I can in fact see that the arg has been passed as `-session-ttl=3.153600e+06`. Any `sessionTTL` within the range documented in the helm chart should not cause the pod to fail to start.

## To Reproduce

`helm install --values headlamp/values.yaml` with a large `sessionTTL` value like:

```
config:
  sessionTTL: 3153600
```

## Environment (please provide info about your environment):
 - Installation type: Helm (k0s)
 - Headlamp Version: 0.44.0
 - Other: helm version: `version.BuildInfo{Version:"v4.2.4", GitCommit:"3900f434fd3ef2b84065dc04508df48f288dba00", GitTreeState:"clean", GoVersion:"go1.26.5", KubeClientVersion:"v1.36"}`

## Are you able to fix this issue?

if needed. it seems like this might be a helm issue, but could be mitigated by editing the chart to allow for string values for `sessionTTL` so that they get passed through without modification.

### PR DESCRIPTION
**Pull request by verdhanyash**
## Summary

Fixes #7325.
Large `sessionTTL` values supplied through a Helm values file can be rendered in scientific notation, causing the Headlamp backend to fail when parsing the `-session-ttl` flag.
This change explicitly converts `sessionTTL` to an integer before rendering the container argument.

## Related Issue

Fixes #7325.

## Changes

- Cast `sessionTTL` to an integer in the Helm deployment template (`charts/headlamp/templates/deployment.yaml`).
- Add a regression test covering the documented maximum value `31536000` (`charts/headlamp/tests/test_cases/session-ttl.yaml`).
- Add expected template fixture for regression test (`charts/headlamp/tests/expected_templates/session-ttl.yaml`).
- Verify that large values are rendered as normal integers rather than scientific notation.

## How to Reproduce the Bug (Before Fix)

1. Create a values file with a large `sessionTTL`:
```yaml
config:
  sessionTTL: 3153600
```
2. Render the chart:
```bash
helm template headlamp ./charts/headlamp --values values.yaml | grep session-ttl
```
3. **Observed (Bug)**: Rendered in scientific notation:
```text
- "-session-ttl=3.1536e+06"
```
When the container starts, the Go flag parser fails with: `invalid value "3.1536e+06" for flag -session-ttl: parse error`.

## Steps to Verify the Fix

1. Provide a Helm values file containing:
```yaml
config:
  sessionTTL: 31536000
```
2. Render the chart:
```bash
helm template headlamp ./charts/headlamp --values <values-file> | grep session-ttl
```
3. **Expected (Fixed)**: Rendered as a standard integer without scientific notation:
```text
- "-session-ttl=31536000"
```

## Tests Ran

- `helm lint ./charts/headlamp`
- `make helm-template-test`
- `git diff --check`

Verified locally:
- `86400` → `-session-ttl=86400`
- `3153600` → `-session-ttl=3153600`
- `31536000` → `-session-ttl=31536000`

## Notes for the Reviewer

- In Helm, numbers from values files are unmarshaled as `float64`. Go template formats `float64` values $\ge 1,000,000$ in scientific notation (`%g`). Using `{{ int .Values.config.sessionTTL }}` casts the value to an integer before string interpolation.
- The default `config.sessionTTL` (`86400`) and all existing chart fixtures remain unchanged.

### CODE REVIEW

File: charts/headlamp/tests/expected_templates/session-ttl.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+﻿---
```
[the remark this answers is not in the document]
Comment (verdhanyash) [reply charts/headlamp/tests/expected_templates/session-ttl.yaml]: fixed it

File: charts/headlamp/templates/deployment.yaml [hunk 2]
```diff
@@ -280,7 +280,7 @@ spec:
             - "-plugins-dir={{ . }}"
             {{- end }}
             {{- if hasKey .Values.config "sessionTTL" }}
-            - "-session-ttl={{ .Values.config.sessionTTL }}"
+            - "-session-ttl={{ int .Values.config.sessionTTL }}"
```
[the remark this answers is not in the document]
Comment (verdhanyash) [reply charts/headlamp/templates/deployment.yaml:283]: Fixed by rebasing onto the latest main: the unrelated base-image digest updates no longer appear in this PR diff, so this cannot re-trigger here.

File: charts/headlamp/tests/expected_templates/session-ttl.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+# Source: headlamp/templates/serviceaccount.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: headlamp
+  namespace: default
+  labels:
+    helm.sh/chart: headlamp-0.44.0
```
[the remark this answers is not in the document]
Comment (verdhanyash) [reply charts/headlamp/tests/expected_templates/session-ttl.yaml]: Regenerated the fixture from helm template output using Helm v3.18.4 against the current chart version 0.45.0, matching what CI renders.

File: charts/headlamp/templates/deployment.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (verdhanyash) [reply charts/headlamp/templates/deployment.yaml:283]: Amended the commit body to record the rationale per the repo commit-message policy: Helm unmarshals values as float64 and Go templates render large float64 in scientific notation, which the backend flag parser rejects.
